sqlmesh/core/engine_adapter/base.py

```diff
@@ -67,6 +68,32 @@ def is_insert_overwrite(self) -> bool:
     def is_replace_where(self) -> bool:
         return self == InsertOverwriteStrategy.REPLACE_WHERE
 
+    @property
+    def requires_condition(self) -> bool:
+        return self.is_replace_where or self.is_delete_insert
+
+
+@dataclass
```

Comment:
Why not pydantic model?

Reply:
I originally had that but then realized it didn't matter since wasn't going to be serialized. I can switch it back though.